libs/core/config/include/hpx/config/lambda_capture_this.hpp

```diff
@@ -0,0 +1,16 @@
+//  Copyright (c) 2021 Srinivas Yadav
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/config/defines.hpp>
+#include <hpx/preprocessor/identity.hpp>
+
+#if defined(HPX_HAVE_CXX20_STD_LAMBDA_CAPTURE)
+#define HPX_CXX20_CAPTURE_THIS(...) HPX_PP_IDENTITY(__VA_ARGS__, this)
```

Comment:
@msimberg yes, that is actually enough.
So, do you suggest changing it to `__VA_ARGS__, this` ?.
If yes, shall I remove only use of `HPX_PP_IDENDTITY` or even the implementation of it in preprocessor module i.e this [file](https://github.com/STEllAR-GROUP/hpx/blob/d41b84d6d4fb0129fb181cbabfc573f77b8dd88f/libs/core/preprocessor/include/hpx/preprocessor/identity.hpp) as well?

Reply:
Wouldn't just writing `__VA_ARGS__, this` leave the comma in place if `__VA_ARGS__` is empty (no arguments)?

Reply:
Good point. It probably does that. Then the question is if `HPX_CXX20_CAPTURE_THIS()` should be an error or not? It would be an error with C++17 and lower with either implementation. With C++20 using `HPX_PP_IDENTITY(__VA_ARGS__, this)` would compile. With C++20 using only `__VA_ARGS__, this` would be an error.

Reply:
When using the identity macro
- a plain `HPX_CXX20_CAPTURE_THIS()` would expand to a simple `this`, which is fine for any C++ version
- `HPX_CXX20_CAPTURE_THIS(=)` would expand to `=, this` for C++20 and `=` for anything below - which is correct as well
- `HPX_CXX20_CAPTURE_THIS(something)` would expand to `something, this` for C++20 and just `something` everywhere else. 
 
The latter should be fine if `something` doesn't have `this` in it too. I think this is the best we can do.

Reply:
Given that the macro is called `HPX_CXX20_CAPTURE_THIS` I think it's definitely problematic if `this` is not captured. However, I also agree with you that there's no perfect solution for this. Enough bikeshedding ;) Let's keep this definition as it is. Our various CI configurations will catch at least catch the trivial cases anyway.